File: Backend/ssaeng-go-jip/src/main/java/com/ssaenggojip/station/entity/Station.java
package com.ssaenggojip.station.entity;

import com.ssaenggojip.common.enums.Line;
import jakarta.persistence.*;
import lombok.*;

@Data
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Entity
@Table(name = "station")
public class Station {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long communityId;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String lineName;

    @Column(nullable = false)
    private Double longitude;

    @Column(nullable = false)
    private Double latitude;
}

File: Frontend/src/types/station.ts
// 공통 응답 타입 (이미 있다면 재사용)
export interface CommonResponse<T> {
  isSuccess: boolean
  code: string
  message: string
  result: T
}

// 전체 역 정보 타입
export interface Station {
  id: number
  name: string
  line: string // "2호선", "신분당선" 등 (문자열 1개)
  latitude: number
  longitude: number
}

// 전체 역 리스트 응답 타입
export type StationListResponse = CommonResponse<Station[]>
